qiskit_experiments/calibration/experiments/drag.py

```diff
@@ -0,0 +1,438 @@
+
+        if reps is None:
+            reps = [1, 3, 5]
+
+        if len(reps) != 3:
+            raise CalibrationError(
+                "As longs as analysis series cannot be dynamically updated "
+                f"{self.__class__.__name__} must use exactly three repetition numbers. "
+                f"Received {reps} with length {len(reps)} != 3."
+            )
+
+        super().set_experiment_options(reps=reps, **fields)
+
+    def __init__(self, qubit: int):
+        """
+        Args:
+            qubit: The qubit for which to run the Drag calibration.
+        """
+
+        super().__init__([qubit])
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+        """Create the circuits for the Drag calibration.
+
+        Args:
+            backend: A backend object.
+
+        Returns:
+            circuits: The circuits that will run the Drag calibration.
+
+        Raises:
+            CalibrationError:
+                - If the beta parameters in the xp and xm pulses are not the same.
+                - If either the xp or xm pulse do not have at least one Drag pulse.
+                - If the number of different repetition series is not three.
+        """
+        # TODO this is temporary logic.
+        self.set_analysis_options(
+            data_processor=get_to_signal_processor(
+                meas_level=self.run_options.meas_level,
+                meas_return=self.run_options.meas_return,
+            ),
+        )
+
+        xm = self.experiment_options.xm
+        xp = self.experiment_options.xp
+
+        if xp is None:
+            beta = Parameter("β")
+            with pulse.build(backend=backend, name="xp") as xp:
+                pulse.play(
+                    pulse.Drag(
+                        duration=self.experiment_options.duration,
+                        amp=self.experiment_options.amp,
+                        sigma=self.experiment_options.sigma,
+                        beta=beta,
+                    ),
+                    pulse.DriveChannel(self._physical_qubits[0]),
+                )
+
+            with pulse.build(backend=backend, name="xm") as xm:
+                pulse.play(
+                    pulse.Drag(
+                        duration=self.experiment_options.duration,
+                        amp=-self.experiment_options.amp,
+                        sigma=self.experiment_options.sigma,
+                        beta=beta,
+                    ),
+                    pulse.DriveChannel(self._physical_qubits[0]),
+                )
+
+        # Now check that we are dealing with Drag pulses.
+        for schedule in [xp, xm]:
+            for block in schedule.blocks:
+                if isinstance(block, pulse.Play):
+                    if isinstance(block.pulse, pulse.Drag):
+                        break
+            else:
+                raise CalibrationError(f"No Drag pulse found in {schedule.name}.")
+
+        beta_xp = next(iter(xp.parameters))
```

Comment:
I think you can use `schedule.get_parameter("beta")` and fix the name of parameter (of make this experiment option) to calibrate to _beta_. User may want to calibrate DRAG subclass or other class to slightly customize the pulse shape.

Reply:
We do not need to do this as the schedule must have one free parameter (and there are checks in place). Doing this will only mean specifying more options making it more annoying to work with this class.